Sources/NIO/BSDSocketAPI.swift

```diff
@@ -49,6 +52,23 @@ import Glibc
 public enum NIOBSDSocket {
 }
 
+extension NIOBSDSocket {
+    /// Specifies the type of socket.
+    public struct SocketType: RawRepresentable {
```

Comment:
Does this need to be public? We don't appear to use it in any public API surface area.

Reply:
mm, I don't remember why I made it public, lets make it internal and then we can change that if needed.